File: server/services/institutions.js
import { API_CONFIG } from '../config/api.js';
import { getAccessToken } from './auth.js';

export async function getInstitutions(country) {
  const token = await getAccessToken();
  const response = await fetch(`${API_CONFIG.baseUrl}/institutions/?country=${country}`, {
    headers: {
      ...API_CONFIG.headers,
      Authorization: `Bearer ${token}`,
    },
  });

  if (!response.ok) {
    throw new Error(`Failed to fetch institutions: ${response.status}`);
  }

  return response.json();
}

File: server/src/services/institutions.py
import httpx

from .auth import get_access_token, API_CONFIG


async def get_institutions(country: str):
    token = get_access_token()

    async with httpx.AsyncClient() as client:
        response = await client.get(
            f"{API_CONFIG['base_url']}/institutions/?country={country}",
            headers={**API_CONFIG["headers"], "Authorization": f"Bearer {token}"},
        )
        response.raise_for_status()
        return response.json()
